### PR TITLE
Add split - grid_size

### PR DESCRIPTION
**Pull request by ttayson**
Adding a grid to divide images

The feature maintains the previous pattern, dividing the image in two, and enables further division into more possibilities.

- [x] Add grid_size function
- [ ] Update Manual

### CODE REVIEW

File: plate_recognition.py [hunk 1]
```diff
@@ -371,18 +381,29 @@ def output_image(args, path, results):
...
     images = [((0, 0), fp)]  # Entire image
 
-    # Top left and top right crops
-    y = 0
-    win_size = 0.55
-    width, height = fp.width * win_size, fp.height * win_size
-    for x in [0, int((1 - win_size) * fp.width)]:
-        images.append(((x, y), fp.crop((x, y, x + width, y + height))))
+    # Determine grid cell size
+    grid_width = fp.width // grid_size[0]
```
Comment (marcbelmont) [inline plate_recognition.py:393]: If you have a crop that splits a license plate in half, we won't read it correctly. Having a bit of overlap between each crop is beneficial. That's why the previous code uses 0.55.

Is the proposed change helping with the new images?

File: plate_recognition.py [hunk 2]
```diff
@@ -241,6 +241,16 @@ def custom_args(parser):
         action="store_true",
         help="Do extra lookups on parts of the image. Useful on high resolution images.",
     )
+    parser.add_argument(
+        "--grid_size_x",
```
Comment (marcbelmont) [inline plate_recognition.py:245]: I would rename it `--split-grid-x` and `--split-grid-y`

Comment (ttayson) [reply plate_recognition.py:245]: ok i will do it

File: plate_recognition.py (hunk 1 above)
Comment (ttayson) [reply plate_recognition.py:393]: @marcbelmont Performing the same tests with both images in a 2x2 setup yielded the expected results; however, the number of calls increased from 3 to 5.

I'm going to add an overlap margin to the function.

File: plate_recognition.py (hunk 2 above)
Comment (marcbelmont) [reply plate_recognition.py:245]: @ttayson Can you wrap this up today?

Comment (ttayson) [reply plate_recognition.py:245]: @marcbelmont , I'm having difficulty determining the expansion area of 0.55. Initially, it's expanding outside the designated area. I'm exploring a logic to capture only the internal area of the images.

As a last resort, I'll expand on all four sides for all cuts.

Another issue is that Python didn't recognize the name you provided, "-split-grid-x"; it interprets it as a split function. Do you have any other suggestions?

Comment (marcbelmont) [reply plate_recognition.py:245]: @ttayson 
That's ok. @danleyb2 is taking over the task as he's been working on something similar before.

**Comment by marcbelmont** [comment]
@ttayson 
Can you please check Brian's version and confirm that it works with the client's image? If everything is good, please merge and share it with the client.

**Review by marcbelmont — APPROVED** [review]

**Comment by ttayson** [comment]
@danleyb2 @marcbelmont , I ran tests and everything worked as expected.

The minimum accepted configuration is split-x=1 and split-y=1.

For a 3x3 grid, we have (3 + 1) * (3 + 1) = 16 images.
For a 2x2 grid, we have (2 + 1) * (2 + 1) = 9 images.
For a 2x1 grid, we have (2 + 1) * (1 + 1) = 6 images.
For a 1x1 grid, we have (1 + 1) * (1 + 1) = 4 images.

Just as a suggestion, a small change from "or" to "and" might be considered, as both parameters are mandatory.

**Pull request merged**